### PR TITLE
feat: add dockerfile

### PR DESCRIPTION
**Pull request by jcam**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM node:erbium-buster AS build
+ENV LANG C.UTF-8
+
+# Copy data for add-on
+COPY . /usr/src/app
+WORKDIR /usr/src/app
+RUN yarn install
```
Comment (Michsior14) [inline Dockerfile:7]:
```suggestion
RUN npm install
RUN npm run build
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+FROM node:erbium-buster-slim
+COPY --from=build /usr/src/app /usr/src/app
+WORKDIR /usr/src/app
+EXPOSE 3000
+USER root
+CMD [ "node_modules/.bin/ts-node", "cli/server.ts", "/dev/ttyUSB0" ]
```
Comment (Michsior14) [inline Dockerfile:14]:
```suggestion
CMD [ "node", "dist/bin/server.js", "/dev/ttyUSB0" ]
```

**Review by Michsior14 — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 node_modules
 package-lock.json
 config.json
+yarn.lock
```
Comment (Michsior14) [inline .gitignore:4]: We don't need yarn.lock here anymore :)

File: .dockerignore [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+*.md
+docs
+Dockerfile
+.vscode
+.nyc_output
+.coverage
+pkg
+test
+node_modules
+yarn.lock
```
Comment (Michsior14) [inline .dockerignore:12]: You could add `package-lock.json` as well.

**Review by Michsior14 — APPROVED** [review]

**Comment by jcam** [comment]
Ah, I was modifying an old version and somehow had my git origins set up wrong so I didn't realize it. Rebasing...

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+RUN npm install
+RUN npm run build
+RUN npm prune --production
+
+FROM node:erbium-buster-slim
+COPY --from=build /usr/src/app /usr/src/app
+WORKDIR /usr/src/app
+EXPOSE 3000
+USER root
+CMD [ "node", "dist/bin/server.js", "/dev/ttyUSB0" ]
```
Comment (balloob) [inline Dockerfile:16]: This makes no sense to hardcode to this path. People need to get their serial stick into Docker.

**Comment by balloob** [comment]
What would be the use of adding this Dockerfile ?

**Comment by jcam** [comment]
It is the dockerfile that I use to generate a package for deploying to my nuc. If not useful, close it out. I was just offering it in case others would find it useful, rather than figuring out how to build it themselves. Could also put it in a docs/examples folder or something

**Comment by Michsior14** [comment]
> It is the dockerfile that I use to generate a package for deploying to my nuc. If not useful, close it out. I was just offering it in case others would find it useful, rather than figuring out how to build it themselves. Could also put it in a docs/examples folder or something

I could also see that happening on my end. I tent to keep zwave controller separately of HassOS.

File: Dockerfile (hunk 5 above)
Comment (balloob) [reply Dockerfile:16]: And also please add a section to the README how to use this Docker file.

**Comment by jcam** [comment]
Closing for now in favor of https://github.com/kpine/dockerfiles/tree/main/zwave-js-server

**Pull request closed without merging**